src/prefect/orion/database/migrations/versions/postgresql/2023_01_26_152801_0a1250a5aa25_expand_work_queue_table.py

```diff
@@ -0,0 +1,365 @@
+                "work_queue_id",
+                prefect.orion.utilities.database.UUID(),
+                nullable=True,
+            )
+        )
+        batch_op.create_index(
+            batch_op.f("ix_deployment__work_queue_id"),
+            ["work_queue_id"],
+            unique=False,
+        )
+        batch_op.create_foreign_key(
+            batch_op.f("fk_deployment__work_queue_id__work_queue"),
+            "work_queue",
+            ["work_queue_id"],
+            ["id"],
+            ondelete="SET NULL",
+        )
+
+    with op.batch_alter_table("work_pool", schema=None) as batch_op:
+        batch_op.create_foreign_key(
+            batch_op.f("fk_work_pool__default_queue_id__work_queue"),
+            "work_queue",
+            ["default_queue_id"],
+            ["id"],
+            ondelete="RESTRICT",
+        )
+        batch_op.alter_column("type", nullable=False)
+
+    with op.batch_alter_table("work_queue", schema=None) as batch_op:
+        batch_op.create_foreign_key(
+            batch_op.f("fk_work_queue__work_pool_id__work_pool"),
+            "work_pool",
+            ["work_pool_id"],
+            ["id"],
+            ondelete="cascade",
+        )
+        batch_op.drop_constraint("uq_work_queue__name")
+        batch_op.create_unique_constraint(
+            op.f("uq_work_queue__work_pool_id_name"), ["work_pool_id", "name"]
+        )
+
+        batch_op.create_index(
+            op.f("ix_work_queue__work_pool_id"),
+            ["work_pool_id"],
+            unique=False,
+        )
+        batch_op.create_index(
+            op.f("ix_work_queue__work_pool_id_priority"),
+            ["work_pool_id", "priority"],
+            unique=False,
+        )
+
+    # Create default agent work pool and associate all existing queues with it
+    connection = op.get_bind()
+
+    connection.execute(
+        sa.text(
+            "INSERT INTO work_pool (name, type) VALUES ('default-agent-pool', 'prefect-agent')"
+        )
+    )
+
+    default_pool_id = connection.execute(
+        sa.text("SELECT id FROM work_pool WHERE name = 'default-agent-pool'")
+    ).fetchone()[0]
+
+    default_queue = connection.execute(
+        sa.text("SELECT id FROM work_queue WHERE name = 'default'")
+    ).fetchone()
+
+    if not default_queue:
+        connection.execute(
+            sa.text(
+                f"INSERT INTO work_queue (name, work_pool_id) VALUES ('default', :default_pool_id)"
+            ).params({"default_pool_id": default_pool_id}),
+        )
+
+    connection.execute(
+        sa.text(
+            "UPDATE work_queue SET work_pool_id = :default_pool_id WHERE work_pool_id IS NULL"
+        ).params({"default_pool_id": default_pool_id}),
+    )
+
+    default_queue_id = connection.execute(
+        sa.text(
+            "SELECT id FROM work_queue WHERE name = 'default' and work_pool_id = :default_pool_id"
+        ).params({"default_pool_id": default_pool_id}),
+    ).fetchone()[0]
+
+    connection.execute(
```

Comment:
I'd break up this data migration into a separate migration, this way we can test to make sure we can do the migration separately in the background

Reply:
I've reduced the scope of this migration to only be schema changes. The data migration is in https://github.com/PrefectHQ/prefect/pull/8327 and the clean up in is #8331